osu.Game/Screens/SelectV2/SongSelect.cs

```diff
@@ -371,8 +382,57 @@ protected override void Update()
                 new Dimension(),
                 new Dimension(GridSizeMode.Relative, 0.5f, minSize: 500, maxSize: 700 + widescreenBonusWidth * 300),
             };
+
+            updateDebounce();
+        }
+
+        #region Selection debounce
+
+        private BeatmapInfo? debounceQueuedSelection;
+        private double debounceElapsedTime;
+
+        private void debounceQueueSelection(BeatmapInfo beatmap)
+        {
+            debounceQueuedSelection = beatmap;
+            debounceElapsedTime = 0;
+        }
+
+        private void updateDebounce()
+        {
+            if (debounceQueuedSelection == null) return;
+
+            // avoid debounce running early if there's a single long frame.
+            debounceElapsedTime += Math.Min(1000 / Clock.FramesPerSecond, Clock.ElapsedFrameTime);
```

Comment:
I'm choosing to ignore the fact that many places in framework choose to leave `Clock.FramesPerSecond` unimplemented by returning 0 (or return 0 as a sort of initial state for the clock), because the semantics of floating point are *for once* nice enough anyway such that the `Math.Min()` call just reduces to returning `Clock.ElapsedFrameTime` in such a case.